File: src/lib/background.cpp
#include <SDL2/SDL.h>
#include <SDL2/SDL_image.h>
#include <string>
#include <sstream>
#include <iostream>
#include "background.h"

void Background::loadTexture(){
  std::stringstream filepath;
  filepath << *ROOT_DIR << "/assets/sprites/";
  filepath << (this->current_mode == DAY ? "background-day.png" : "background-night.png");
  this->surface = IMG_Load(filepath.str().c_str());
}

void Background::render(){
  this->updateFrame();
  SDL_RenderCopy(this->gameRenderer, this->texture, NULL, NULL);
  SDL_DestroyTexture(this->texture);
}

void Background::updateFrame(){
  this->loadTexture(); 
  this->texture = SDL_CreateTextureFromSurface(this->gameRenderer, this->surface);
  SDL_FreeSurface(this->surface);
}

File: src/background.ts
import { type Application, Assets, Sprite, type Texture } from "pixi.js";

export class Background {
  public sprite: Sprite;

  async load(app: Application) {
    const texture: Texture = await Assets.load(
      "/assets/sprites/background-day.png",
    );
    this.sprite = new Sprite(texture);
    this.sprite.anchor.set(0.5);
    this.sprite.position.set(app.screen.width / 2, app.screen.height / 2);
    this.sprite.zIndex = -90;
  }

  constructor(_app: Application) {
    this.sprite = new Sprite();
  }
}
